python/ray/data/dataset.py

```diff
@@ -1853,8 +1857,16 @@ def iter_rows(self, *, prefetch_blocks: int = 0) -> Iterator[T]:
         Returns:
             A local iterator over the entire dataset.
         """
+        dataset_format = self._dataset_format()
+        batch_format = (
+            "pyarrow"
+            if dataset_format == "arrow"
+            else "pandas"
+            if dataset_format == "pandas"
+            else "native"
```

Comment:
Please add a comment "During row-based ops, we also choose a batch format that lines up with the current dataset format in order to eliminate unnecessary copies and type conversions."